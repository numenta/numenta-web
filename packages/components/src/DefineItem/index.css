/**
 * Numenta Web Platform and Sites source code
 * MIT License (see LICENSE.txt)
 * Copyright © 2005—2017 Numenta <http://numenta.com>
 */

.horizontal {
  composes: fl from 'tachyons';
  composes: h-100 from 'tachyons';
  composes: ma0 from 'tachyons';
  padding-bottom: 0.2rem;
  padding-top: 0.4rem;
  width: 65%;
}

.vertical {
  composes: mb3 from 'tachyons';
  composes: ml0 from 'tachyons';
  composes: pl0 from 'tachyons';
}


.defineItem {
  word-wrap: break-word;
}
