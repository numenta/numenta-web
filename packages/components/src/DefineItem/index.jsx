// Numenta Web Platform and Sites source code
// MIT License (see LICENSE.txt)
// Copyright © 2005—2017 Numenta <http://numenta.com>

import classNames from 'classnames'
import React from 'react'

import styles from './index.css'


const DefineItem = ({children, direction}) => {
  const classes = [styles.defineItem]

  if (direction) classes.push(styles[direction])

  return (
    <dd className={classNames(...classes)}>
      {children}
    </dd>
  )
}

DefineItem.propTypes = {
  children: React.PropTypes.node.isRequired,
  direction: React.PropTypes.oneOf(['horizontal', 'vertical']),
}

export default DefineItem
