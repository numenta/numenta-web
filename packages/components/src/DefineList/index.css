/**
 * Numenta Web Platform and Sites source code
 * MIT License (see LICENSE.txt)
 * Copyright © 2005—2017 Numenta <http://numenta.com>
 */

.horizontal {
  composes: cf from 'tachyons';
  composes: lh-title from 'tachyons';
}

.vertical {
  composes: lh-copy from 'tachyons';
  composes: mb3 from 'tachyons';
  composes: measure from 'tachyons';
  composes: measure-wide-l from 'tachyons';
}


.defineList {
  composes: ma0 from 'tachyons';
  composes: pa0 from 'tachyons';
}
