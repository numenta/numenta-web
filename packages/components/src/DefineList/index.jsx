// Numenta Web Platform and Sites source code
// MIT License (see LICENSE.txt)
// Copyright © 2005—2017 Numenta <http://numenta.com>

import classNames from 'classnames'
import React from 'react'

import styles from './index.css'


const DefineList = ({children, direction}) => {
  const classes = [styles.defineList, styles[direction]]
  const childrenWithProps = React.Children.map(children, (child) =>
    React.cloneElement(child, {direction})
  )

  return (
    <dl className={classNames(...classes)}>
      {childrenWithProps}
    </dl>
  )
}

DefineList.propTypes = {
  children: React.PropTypes.node.isRequired,
  direction: React.PropTypes.oneOf(['horizontal', 'vertical']),
}

DefineList.defaultProps = {
  direction: 'vertical',
}

export default DefineList
