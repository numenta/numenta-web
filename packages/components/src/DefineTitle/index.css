/**
 * Numenta Web Platform and Sites source code
 * MIT License (see LICENSE.txt)
 * Copyright © 2005—2017 Numenta <http://numenta.com>
 */

.horizontal {
  composes: fl from 'tachyons';
  composes: h-100 from 'tachyons';
  composes: ma0 from 'tachyons';
  padding-bottom: 0.2rem;
  padding-top: 0.4rem;
  width: 35%;
}

.vertical {
  composes: lh-title from 'tachyons';
  composes: mb1 from 'tachyons';
}


.defineTitle {
  composes: fw6 from 'tachyons';
}
