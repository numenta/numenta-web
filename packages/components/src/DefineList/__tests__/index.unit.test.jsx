import React from 'react'
import renderer from 'react-test-renderer'

import DefineItem from '../../DefineItem'
import DefineList from '../../DefineList'
import DefineTitle from '../../DefineTitle'


describe('DefineList React component', () => {

  it('Renders Default correctly', () => {
    const component = renderer.create(
      <DefineList>
        <DefineTitle>
          Title 1
        </DefineTitle>
        <DefineItem>
          Item 1
        </DefineItem>
        <DefineTitle>
          Title 2
        </DefineTitle>
        <DefineItem>
          Item 2
        </DefineItem>
      </DefineList>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Renders Horizontally correctly', () => {
    const component = renderer.create(
      <DefineList direction="horizontal">
        <DefineTitle>
          Title 1
        </DefineTitle>
        <DefineItem>
          Item 1
        </DefineItem>
        <DefineTitle>
          Title 2
        </DefineTitle>
        <DefineItem>
          Item 2
        </DefineItem>
      </DefineList>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Renders Vertically correctly', () => {
    const component = renderer.create(
      <DefineList direction="vertical">
        <DefineTitle>
          Title 1
        </DefineTitle>
        <DefineItem>
          Item 1
        </DefineItem>
        <DefineTitle>
          Title 2
        </DefineTitle>
        <DefineItem>
          Item 2
        </DefineItem>
      </DefineList>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

})
