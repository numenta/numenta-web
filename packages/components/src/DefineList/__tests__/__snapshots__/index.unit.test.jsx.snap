exports[`DefineList React component Renders Default correctly 1`] = `
<dl
  className="defineList vertical">
  <dt
    className="defineTitle vertical">
    Title 1
  </dt>
  <dd
    className="defineItem vertical">
    Item 1
  </dd>
  <dt
    className="defineTitle vertical">
    Title 2
  </dt>
  <dd
    className="defineItem vertical">
    Item 2
  </dd>
</dl>
`;

exports[`DefineList React component Renders Horizontally correctly 1`] = `
<dl
  className="defineList horizontal">
  <dt
    className="defineTitle horizontal">
    Title 1
  </dt>
  <dd
    className="defineItem horizontal">
    Item 1
  </dd>
  <dt
    className="defineTitle horizontal">
    Title 2
  </dt>
  <dd
    className="defineItem horizontal">
    Item 2
  </dd>
</dl>
`;

exports[`DefineList React component Renders Vertically correctly 1`] = `
<dl
  className="defineList vertical">
  <dt
    className="defineTitle vertical">
    Title 1
  </dt>
  <dd
    className="defineItem vertical">
    Item 1
  </dd>
  <dt
    className="defineTitle vertical">
    Title 2
  </dt>
  <dd
    className="defineItem vertical">
    Item 2
  </dd>
</dl>
`;
